chalice/chalicelib/aws/gds_aws_client.py

```diff
@@ -285,11 +285,12 @@ def assume_chained_role(self, target_account):
         return: bool
         """
         assumed = False
+        chain = self.get_chain_role_params()
```

Comment:
`self.get_chain_role_params` does not return anything, it simply modifies the `chain` (i.e. `self.chain`) attribute of the object.

`chain` here will be `None`. As such, trying to index it like a dictionary (lines 290, 292, 293, and for the same reason lines 309, 311 and 312 in the `get_chained_session` function) will cause a TypeError.

Reply:
You're right. I made that change to return self.chain from get_chain_role_params but somehow didn't commit it.

Reply:
That's so bizarre - I deployed that branch and ran audits before making the PR so I must have reverted the change afterwards - fixed now. Thanks.